src/workerd/io/actor-sqlite.c++

```diff
@@ -202,6 +203,8 @@ void ActorSqlite::onCriticalError(
     kj::StringPtr errorMessage, kj::Maybe<kj::Exception> maybeException) {
   // If we have already experienced a terminal exception, no need to replace it
   if (broken == kj::none) {
+    // TODO(someday): If we are setting the broken exception, do we also need to explicitly break
+    // the output gate?
```

Comment:
I noticed that this "critical error" handler was another place where the ActorSqlite "broken" exception gets set, but it wasn't obvious to me that this would necessarily result in the output gate breaking or the actor getting reset.

Looking through the caller, I guess this only fires for certain [sqlite error conditions during a transaction](https://sqlite.org/c3ref/get_autocommit.html)?